java/configuration/src/test/java/sleeper/configuration/properties/format/GeneratePropertiesTemplatesTest.java

```diff
@@ -0,0 +1,256 @@
+        @Override
+        public Stream<Arguments> provideArguments(ExtensionContext context) {
+            return SystemDefinedInstanceProperty.getAll().stream()
+                    .map(Arguments::of);
+        }
+    }
+
+    @Nested
+    @DisplayName("Generate full example instance properties")
+    class GenerateFullInstanceProperties {
+        private final String propertiesString = loadFileAsString("example/full/instance.properties");
+
+        @ParameterizedTest
+        @ArgumentsSource(MandatoryInstancePropertyTemplateValues.class)
+        void shouldSetMandatoryParameters(UserDefinedInstanceProperty property, String value) {
+            assertThat(instancePropertiesFromString(propertiesString)
+                    .get(property))
+                    .isEqualTo(value);
+        }
+
+        @ParameterizedTest
+        @ArgumentsSource(SystemDefinedInstanceProperties.class)
+        void shouldExcludeSystemDefinedProperties(SystemDefinedInstanceProperty property) {
+            assertThat(propertiesString)
+                    .doesNotContain(property.getPropertyName() + "=");
+        }
+
+        @Test
+        void shouldNotCommentOutUnsetParameter() {
+            assertThat(propertiesString)
+                    .contains(System.lineSeparator() +
+                            "sleeper.userjars=" +
+                            System.lineSeparator());
+        }
+
+        @Test
+        void shouldNotCommentOutParameterSetToDefaultValue() {
+            assertThat(propertiesString)
+                    .contains(System.lineSeparator() +
+                            "sleeper.stack.tag.name=DeploymentStack" +
+                            System.lineSeparator());
+        }
+
+        @Test
+        void shouldExcludeTagsProperty() {
+            assertThat(propertiesString)
+                    .doesNotContain("sleeper.tags=");
+        }
+    }
+
+    @Nested
+    @DisplayName("Generate full example table properties")
+    class GenerateFullTableProperties {
+        private final String propertiesString = loadFileAsString("example/full/table.properties");
+
+        @Test
+        void shouldGenerateValidTableProperties() {
+            assertThat(tablePropertiesFromString(propertiesString)
+                    .get(TABLE_NAME))
+                    .isEqualTo("example-table");
+        }
+
+        @Test
+        void shouldNotSetSchemaInFile() {
+            assertThat(propertiesString)
+                    .doesNotContain(SCHEMA.getPropertyName());
+        }
+    }
+
+    @Nested
+    @DisplayName("Generate basic example instance properties")
+    class GenerateBasicInstanceProperties {
+        private final String propertiesString = loadFileAsString("example/basic/instance.properties");
+
+        @Test
+        void shouldGenerateValidInstanceProperties() {
+            assertThat(instancePropertiesFromString(propertiesString)
+                    .get(ID))
+                    .isEqualTo("full-example");
```

Comment:
It's a bit strange how the basic instance.properties example uses the instance ID of `full-example`, perhaps we could use `basic-example` here?

Reply:
Done, thanks